src/PhotoNim/shapes.nim

```diff
@@ -5,142 +5,196 @@ import std/options
 import camera, geometry
 
 
-type 
+type HitRecord* = object
+    ray*: Ray
+    t_hit*: float32
+    world_pt*: Point3D
+    surface_pt*: Point2D
```

Comment:
I like this shapes.nim design more... I feel like it's better to define HitRecord first considering that we have to use it in rayIntersection procedure